gsl/span

```diff
@@ -625,6 +625,34 @@ as_writeable_bytes(span<ElementType, Extent> s) noexcept
     return {reinterpret_cast<byte*>(s.data()), s.size_bytes()};
 }
 
+//
+// make_span() - Utility functions for creating spans
+//
+template <class ElementType>
+span<ElementType>
+make_span(ElementType *ptr, typename span<ElementType>::index_type &&count)
+{ return span<ElementType>(ptr, std::forward<typename span<ElementType>::index_type>(count)); }
+
+template <class ElementType>
+span<ElementType>
+make_span(ElementType *firstElem, ElementType *lastElem)
+{ return span<ElementType>(firstElem, lastElem); }
+
+template <class ElementType, size_t N>
+span<ElementType>
+make_span(ElementType (&arr)[N])
+{ return span<ElementType>(arr); }
+
+template <class Container>
+span<typename Container::value_type>
+make_span(Container &cont)
+{ return span<typename Container::value_type>(cont); }
+
+template <class Container>
+span<const typename Container::value_type>
+make_span(const Container &cont)
+{ return span<const typename Container::value_type>(cont); }
+
```

Comment:
Do you think we should add overloads for `unique_ptr` and `shared_ptr` now we've added those ctors?